sidecar_acceptor.go

```diff
@@ -601,6 +601,13 @@ func (a *SidecarAcceptor) matchPrepare(pendingBatch *order.Batch,
 	sdcrLog.Infof("Received PrepareMsg for batch=%x, num_orders=%v",
 		batch.ID[:], len(batch.MatchedOrders))
 
+	// Ensure that we do not have any registered shims for the orders
+	// in this batch. This is not supposed to happen but we have a bug.
+	if err = a.removeShims(batch); err != nil {
```

Comment:
Perhaps link to the PR where this will be fixed for good?
Also add a comment that `removeShims` will only return an error for serious problems (for example not being able to read orders from disk) and not if there is no pending shim (only a warning will be logged but no error returned).